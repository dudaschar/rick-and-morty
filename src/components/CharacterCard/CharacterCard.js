import React from 'react';
import PropTypes from 'prop-types';

import { Card, Content } from './styles/Wrapper';
import { Image } from './styles/Image';
import { CardTitle } from './styles/Typography';

function CharacterCard({ info }) {
  return (
    <Card href={`/list/details?character=${info.id}`}>
      <Image src={info.image} />
      <Content>
        <CardTitle>{info.name}</CardTitle>
        <p><strong>Episodes:</strong> {info.episode.length}</p>
        <p><strong>Last location:</strong> {info.location.name}</p>
      </Content>
    </Card>
  );
}

CharacterCard.propTypes = {
  info: PropTypes.object.isRequired,
}

export default CharacterCard;
