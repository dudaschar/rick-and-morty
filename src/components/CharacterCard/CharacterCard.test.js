import React from 'react';
import { render } from 'src/testUtils/testUtils';

import CharacterCard from './CharacterCard';

describe('CharacterCard Component', () => {
  it('should render the component properly', () => {
    const props = {
      id: 1,
      image: 'Rick_image_path.jpg',
      name: 'Rick',
      episode: [
        'episode 1',
      ],
      location: {
        name: 'Earth',
      }
    }

    const { asFragment } = render(
      <CharacterCard
        info={props}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});