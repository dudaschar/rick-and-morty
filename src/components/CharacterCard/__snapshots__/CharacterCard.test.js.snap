// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard Component should render the component properly 1`] = `
<DocumentFragment>
  <a
    class="Wrapper__Card-f2s50d-0 jLWtYJ"
    href="/list/details?character=1"
  >
    <img
      class="Image-sc-1012c2w-0 gGBmtk"
      src="Rick_image_path.jpg"
    />
    <div
      class="Wrapper__Content-f2s50d-1 hFMuHc"
    >
      <p
        class="Typography__CardTitle-lx9mxg-0 igOmMr"
      >
        Rick
      </p>
      <p>
        <strong>
          Episodes:
        </strong>
         1
      </p>
      <p>
        <strong>
          Last location:
        </strong>
         Earth
      </p>
    </div>
  </a>
</DocumentFragment>
`;
